docs/docs/api/Strapi.mdx

```diff
@@ -0,0 +1,525 @@
+toc_max_heading_level: 3
+---
+import Type from '@site/docs/api/components/type';
+
+# Strapi
+
+:::info
+
+Current state: **Stable**
+
+:::
+
+The Strapi class is the main object used in Strapi projects.
+An instance of Strapi class is available as a global in any Strapi project: `global.strapi`.
+
+## Class: Strapi
+
+### `new Strapi(opts)`
+
+- `opts`: <Type>Object</Type> Options that can be used on Strapi startup
+  - `autoReload`: <Type>Boolean</Type> **Default:** true
+    - If false, deactivate auto reload
+    - If you modify any file in your Strapi project, it reloads your nodejs app
+    - If any content-type is changed, it will reload the nodejs app
+  - `serveAdminPanel`: <Type>Boolean</Type> **Default:** true
+    - Should the admin panel be loaded and serve as a web client
+    - The admin panel build will not be delivered if false
+  - `appDir`: <Type>String</Type>  **Default:** `process.cwd()`
+    - The directory relative or absolute path where Strapi will write every file (schemas, generated APIs, controllers or services)
+  - `distDir`: <Type>String</Type>  **Default:** appDir value
+    - The directory relative or absolute path where Strapi will read configurations, schemas and any compiled code
+
+Instances of the Strapi class can be created using the new keyword.
+
+```javascript
+const strapiInstance = new Strapi();
+```
+
+### `strapi.container`
+- [<Type>Container</Type>](./container)
+
+This is where all registries are stored.
+
+See [Container](./container).
+
+### `strapi.dirs`
+- <Type>Object</Type>
+
+Stored paths of file system.
+
+- `dirs.dist`: <Type>[StrapiPathObject](#strapipathobject)</Type>
+  - Build folder
+- `dirs.app`: <Type>[StrapiPathObject](#strapipathobject)</Type>
+  - Sources folder
+- `dirs.static`: <Type>Object</Type> Define path to directories involving web client display
+  - `public`:  <Type>String</Type> Path to the folder to serve publicly (like files, images, etc..)
+
+#### StrapiPathObject
+- <Type>Object</Type>
+
+A set of path to specific Strapi project parts.
+
+- `root`: <Type>String</Type> Root path
+- `src`: <Type>String</Type> Sources route path to project files
+- `api`:  <Type>String</Type> Path to the folder containing project developers' API files (content-types, controllers, services, routes, etc..)
+- `components`:  <Type>String</Type> Path to the folder containing project developers' components
+- `policies`:  <Type>String</Type> Path to the folder where the Strapi project developers' policies are stored
+  - A set of functions that check the state of the data and prevent the access to the API accordingly
+- `middlewares`:  <Type>String</Type> Path to the folder where the Strapi project developers' middlewares are stored
+  - A set of function that wrap around routes and requests
+- `config`:  <Type>String</Type> Path to the folder containing project developers' config files
+
+### `strapi.isLoaded`
+- <Type>Boolean</Type>
+
+  - `true`: Everything have been loaded
```

Comment:
```suggestion
  - `true`: Everything has been loaded
```

What is everything? The app? the lifecycles? is this true when the http server is already running, or before?